### ISSUE TITLE
add Tooling API example to README?

### ISSUE THREAD

**Issue by chrismcmahon** [ejholmes/restforce#268]
An example in the README of how to use restforce with the Tooling API would be helpful. I’ve been using restforce with the regular API for over a year now, but I’m struggling to figure out how to add a CustomField to an Account object with restforce and the Tooling API.

**Comment by timrogers** [comment]
@chrismcmahon This would be a great idea but I'm not personally familiar with the Tooling API. I'll try to take a look myself. Have you had a delve into the code?

**Comment by chrismcmahon** [comment]
I have. It seems to me that something like the following should work, but all the tweaks I've tried return 404 

  @tooling_api_client ||= Restforce.tooling api_version: '32.0',
                                            refresh_token: ENV['SF_REFRESH_TOKEN'],
                                            client_id: ENV['SF_CLIENT_KEY'],
                                            client_secret: ENV['SF_CLIENT_SECRET']

Restforce.log = true

  @tooling_api_client.create!('Account', CustomField: {
      DeveloperName: 'PreviousLevel__c', Metadata: {
       label: "Previous Level",
       name: "PreviousLevel__c",
       referenceTo: ["npsp__Level__c"],
       relationshipName: "PreviousLevel__r",
       type: "reference"
    }
  }
  )

**Comment by chrismcmahon** [comment]
This is interesting. I set Restforce.log=true and this is what restforce is sending to the Tooling API. I don't think this is what's intended: 

D, [2016-11-10T11:59:10.636615 #95103] DEBUG -- request:
  url: "https://cs21.salesforce.com/services/data/v32.0/tooling/sobjects/Account"
  method: :post
  headers: {"User-Agent"=>"Faraday v0.9.2", "Content-Type"=>"application/json", "Authorization"=>"OAuth <REDACTED>"}
  body: "{\"CustomField\":{\"DeveloperName\":\"PreviousLevel__c\",\"Metadata\":{\"label\":\"Previous Level\",\"name\":\"PreviousLevel__c\",\"referenceTo\":[\"npsp__Level__c\"],\"relationshipName\":\"PreviousLevel__r\",\"type\":\"reference\"}}}"

**Comment by timrogers** [comment]
@chrismcmahon Any ideas on a fix? It could just be broken potentially - I haven't seen others using the Tooling API.

**Comment by chrismcmahon** [comment]
I think I just don't know enough about the Tooling API nor about restforce. I've treated restforce so far as basically a black box.  I'll dig into this more next week with a colleague who knows a lot more about SF and restforce, but in the meantime, even the simplest working example would be helpful.

**Comment by timrogers** [comment]
@chrismcmahon Makes sense to me. I'd like to look into this a bit more but I'm too busy at the moment. Happy to investigate at some point!

**Comment by kevinrobell-st** [comment]
This can also probably be closed now.
